### ISSUE TITLE
PT2.1 cudagraph trees loss divergence

### ISSUE THREAD

**Issue by ErnestChan** [pytorch/pytorch#118228]
### 🐛 Describe the bug

On Pytorch 2.1, we're training a model with reduce-overhead and cudagraph trees turned on. It gives us a large speedup but the loss curves diverge from both eager and default compile. Loss curves match between eager mode and default compile, so it seems the issue is in cuda graphing.

I tried the accuracy minifier and was able to create a repro that throws an AccuracyError, but I don't think the repro uses cuda graphs. Also the repro is not very mini, has thousands of lines of aten ops. Could still be a place to start
[repro.py.txt](https://github.com/pytorch/pytorch/files/14043863/repro.py.txt)

### Versions

Ubuntu 22.04.1 LTS
Pytorch 2.1 with a few custom and upstream patches

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @mcarilli @eellison @peterbell10 @bdhirsh @anijain2305 @chauhang

**Comment by ezyang** [comment]
cc @eellison

**Comment by xw285cornell** [comment]
Just wondering, does the issue exist with cudagraph but no cudagraph tree? This may help narrow down the issue though it may use more memory

**Comment by jansel** [comment]
@eellison to provide update

**Comment by eellison** [comment]
cc @ErnestChan - we've made a lot of improvements since. Do you have anyway of verifying if this exists on main?

**Comment by eellison** [comment]
per @ErnestChan , okay to close.
